Stackchain/alonzo metadata updates

stackchain opened the pull request.

v-almonacid commented on rust/src/android/transaction_metadata.rs:
```diff
@@ -41,30 +44,34 @@ pub unsafe extern "C" fn Java_io_emurgo_rnhaskellshelley_Native_transactionMetad
 #[no_mangle]
-pub unsafe extern "C" fn Java_io_emurgo_rnhaskellshelley_Native_transactionMetadataNew(
-  env: JNIEnv, _: JObject, general_ptr: JRPtr
+pub unsafe extern "C" fn Java_io_emurgo_rnhaskellshelley_Native_txAuxiliaryDataNew(
```
I'd suggest you call this `auxiliaryDataNew`. It's not technically required but all function names, across all languages, respect a strict naming convention.

In this case, for a rust function called from java, it's `Java_io_emurgo_rnhaskellshelley_Native_structNameMethodName`.

I think it's important to keep a consistent mapping between layers. One advantage is that, when you are writing the Java function that calls this rust function, you _know_ that it's called `auxiliaryDataNew(...)`, and you don't need to check what name you picked in the rust module. In the long run, these small details will save you some time and bugs.

v-almonacid commented on rust/src/ios/transaction_metadata.rs:
```diff
@@ -8,50 +8,55 @@ use super::data::DataPtr;
 
 #[no_mangle]
-pub unsafe extern "C" fn transaction_metadata_from_bytes(
+pub unsafe extern "C" fn tx_auxiliary_data_from_bytes(
```
Similarly, functions in this module should be called `auxiliary_data_method_name` (and the module itself, `auxiliary_data.rs` instead of `transaction_metadata.rs`).

v-almonacid commented:
Hey @stackchain, you requested a review and marked this PR as ready, however, it only touches rust code and merging as it is will break a few things.

A few issues I'd address before marking this a ready:
- Complete the remaining wrappers in JS/Java/Objective-C.
- In `cargo.toml`, you need to point to the alonzo branch in `cardano_serialization-lib`, since you are referring to new rust types that do not exist in the current version.
- Add some simple tests in `example/`. This is very rudimentary but it's the only way I came up with since AFAIK we can't test using jest unit tests.
- I'd suggest that the `rust/src/[android|ios]/transaction_metadata.rs` modules are renamed as `auxiliary_data.rs`, since the name of the module refers exactly to the struct it creates bindings for (but this is just a convention as I explained previously).

Also, I changed the base branch from `master` to `alonzo` (which at the moment is equivalent to master).

stackchain commented:
> Hey @stackchain, you requested a review and marked this PR as ready, however, it only touches rust code and merging as it is will break a few things.
> 
> A few issues I'd address before marking this a ready:
> 
>     * Complete the remaining wrappers in JS/Java/Objective-C.
> 
>     * In `cargo.toml`, you need to point to the alonzo branch in `cardano_serialization-lib`, since you are referring to new rust types that do not exist in the current version.
> 
>     * Add some simple tests in `example/`. This is very rudimentary but it's the only way I came up with since AFAIK we can't test using jest unit tests.
> 
>     * I'd suggest that the `rust/src/[android|ios]/transaction_metadata.rs` modules are renamed as `auxiliary_data.rs`, since the name of the module refers exactly to the struct it creates bindings for (but this is just a convention as I explained previously).
> 
> 
> Also, I changed the base branch from `master` to `alonzo` (which at the moment is equivalent to master).



> Hey @stackchain, you requested a review and marked this PR as ready, however, it only touches rust code and merging as it is will break a few things.
> 
> A few issues I'd address before marking this a ready:
> 
>     * Complete the remaining wrappers in JS/Java/Objective-C.
> 
>     * In `cargo.toml`, you need to point to the alonzo branch in `cardano_serialization-lib`, since you are referring to new rust types that do not exist in the current version.
> 
>     * Add some simple tests in `example/`. This is very rudimentary but it's the only way I came up with since AFAIK we can't test using jest unit tests.
> 
>     * I'd suggest that the `rust/src/[android|ios]/transaction_metadata.rs` modules are renamed as `auxiliary_data.rs`, since the name of the module refers exactly to the struct it creates bindings for (but this is just a convention as I explained previously).
> 
> 
> Also, I changed the base branch from `master` to `alonzo` (which at the moment is equivalent to master).

Yap, I renamed the branch from `alonzo` to `alonzo-test` to follow the Rob pattern on the lib, gonna updating and testing on mobile today.

stackchain commented:
- [ ] Rename `transaction_metadata` to `auxiliary_data`
- [ ] Update Cargo.toml to consume the `alonzo-test` branch of `cardado-serialization-lib`
- [ ] Add tests
